xbmc/utils/StringUtils.cpp

```diff
@@ -1810,3 +1810,21 @@ const std::locale& StringUtils::GetOriginalLocale() noexcept
 {
   return g_langInfo.GetOriginalLocale();
 }
+
+void StringUtils::RemoveBetween(std::pair<char, char> bounderies,
+                                char charToRemove,
+                                std::string& str)
+
+{
+  int skip = 0;
+  str.erase(std::remove_if(str.begin(), str.end(),
```

Comment:
to avoid pushing so much text so far right, you can separate this into two lines, something like:

```c++
auto shouldRemove = [&](char c)
  {
    if (c == bounderies.first)
      skip++;
    else if (c == bounderies.second && skip)
      skip--;

    return (skip && c == charToRemove);
  };

str.erase(std::remove_if(str.begin(), str.end(), shouldRemove), str.end());
```

Reply:
thanks @garbear yes, that's far better. let's see if it's accepted in general